homeassistant/components/foscam/__init__.py

```diff
@@ -81,3 +98,38 @@ def update_unique_id(entry):
     LOGGER.info("Migration to version %s successful", entry.version)
 
     return True
+
+
+class FoscamCoordinator(DataUpdateCoordinator):
+    """My custom coordinator."""
+
+    def __init__(self, hass, session, update_interval=timedelta(seconds=30)):
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            LOGGER,
+            # Name of the data. For logging purposes.
+            name=DOMAIN,
+            # Polling interval. Will only be polled if there are subscribers.
+            update_interval=update_interval,
+        )
+        self._session = session
+
+    async def _async_update_data(self):
+        """Fetch data from API endpoint."""
+        try:
+            async with async_timeout.timeout(30):
+                data = {}
+                ret, dev_info = await self.hass.async_add_executor_job(
+                    self._session.get_dev_info
+                )
+                if ret == 0:
+                    data["dev_info"] = dev_info
+                data["product_info"] = (
+                    await self.hass.async_add_executor_job(
+                        self._session.get_product_all_info
+                    )
+                )[1]
+                return data
+        except Exception as err:
```

Comment:
Should have the correct possible exceptions and not just catch broad

Reply:
I think this was copy paste, no error should be thrown as far as I'm aware, removing the try-except